src/js/midi/helpers.js

```diff
@@ -1,5 +1,16 @@
-const getNameFromPort = (port) => `${port.name} (version ${port.version}) ${port.manufacturer}`
+const getNameFromPort = (port) => {
+  const { id, name, version, manufacturer } = port
+  return `${name} (version ${version}) ${manufacturer} (${id})`
+}
 
-const getAllKeys = () => R.unfold((n) => (n > keyIdMax ? false : [n, n + 1]), keyIdMin)
-const getWhiteKeys = R.compose(R.map(parseInt), R.keys, () => whiteOnlyMap)
-const getBlackKeys = R.converge(R.difference, [getAllKeys, getWhiteKeys])
+const getAllKeys = () => {
+  return [...Array(128).keys()] // [0, 1, 2, ..., 127]
+}
+
+const getWhiteKeys = () => {
+  return Object.keys(whiteOnlyMap).map((id) => parseInt(id))
```

Comment:
I might be a good idea to pre-compute this. It's called on each note event and could become a performance issue.